### PR TITLE
Refactoring the argparser and the config

### PR DESCRIPTION
**Pull request by Bishoy-at-pieces**

### CODE REVIEW

File: src/pieces/app.py [hunk 1]
```diff
@@ -1,147 +1,124 @@
...
+        commit_parser = self.command_parser.add_parser('commit', help='Auto generate a github commit messaage and commit changes')
+        commit_parser.add_argument("-p","--push",dest="push",action="store_true", help="push the code to github")
+        commit_parser.set_defaults(func=git_commit)
+
+    def run(self):
+        try:
+            arg = sys.argv[1]
+        except IndexError: # No command provided
+            print_help()
+            return
```
Comment (hal-8999-alpha) [inline src/pieces/app.py:113]: Any speed improvements? Or just for readability?

File: src/pieces/commands/cli_loop.py [hunk 2]
```diff
@@ -11,41 +11,7 @@
...
-        previous_row = current_row
-    
-    # The last element of the previous row contains the levenshtein distance
-    return previous_row[-1]
-
-
-def find_most_similar_command(valid_commands, user_input):
-    # Calculate the Levenshtein distance between the user input and each valid command
-    distances = {cmd: levenshtein_distance(user_input, cmd) for cmd in valid_commands}
-    # Find the command with the smallest Levenshtein distance to the user input
```
Comment (hal-8999-alpha) [inline src/pieces/commands/cli_loop.py:46]: Very cool

**Review by hal-8999-alpha — APPROVED** [review]
Looks good to me. If we run into any issues in the run command we may want to revisit here but should be good. Great work!

**Review by caleb-at-pieces — APPROVED** [review]
good cleanup

**Review by hal-8999-alpha — APPROVED** [review]
Huge refactor, great work!

**Review by mark-at-pieces — COMMENTED** [review]
LGTM

**Pull request merged**